engine/glfw/lib/android/android_util.c

```diff
@@ -123,10 +124,39 @@ static EGLint choose_egl_config(EGLDisplay display, EGLConfig* config)
     return result;
 }
 
+static int IsAppAndWindowReady(_GLFWwin_android* win)
+{
+    return win != 0 && win->app != 0 && win->app->window != 0;
+}
+
+static int WaitForAppAndWindow(_GLFWwin_android* win)
+{
+    useconds_t  wait_period = 50*1000;
+    int         num_waits = 10;
+    do {
+        if (IsAppAndWindowReady(win)) {
+
+            LOGI("ENGINE THREAD: Window ready!");
+            return 1;
+        }
+        LOGI("ENGINE THREAD: Window not ready. Waiting...");
+        usleep(wait_period);
+    } while(--num_waits > 0);
+
+    LOGI("ENGINE THREAD: Window not ready. Exiting!");
+    return 0;
+}
+
 int init_gl(_GLFWwin_android* win)
 {
     LOGV("init_gl");
 
+    if (!WaitForAppAndWindow(win))
+    {
+        LOGE("ENGINE THREAD: Window not ready. Returning from init_gl()");
+        return 0;
```

Comment:
If we still don't have a window, we don't want it to continue (it's what currently generates the crashes).

Instead we return 0 here, which should exit the game.
I think this is preferable to a crash.